src/credentials/providers/sharedCredentialsProvider.ts

```diff
@@ -189,8 +189,20 @@ export class SharedCredentialsProvider implements CredentialsProvider {
             logger.verbose(
                 `Profile ${this.profileName} contains ${SHARED_CREDENTIAL_PROPERTIES.CREDENTIAL_PROCESS} - treating as Process Credentials`
             )
+            // Guard against hangs: credential_process will wait forever if the process does not return a result.
```

Comment:
is there any chance we will need this pattern for other profile types? Should we lift this into a function? That may also make this easier to follow (encapsulates/isolates the workaround).

Reply:
There's a decent possibility of other providers never returning from `refresh`. It really just depends on how they are trying to get credentials.